drivers/timer/smartbond_timer.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2023 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/timer/system_timer.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/spinlock.h>
+#include <cmsis_core.h>
+#include <zephyr/irq.h>
+#include <da1469x_pdc.h>
+
+#define COUNTER_SPAN BIT(24)
+#define CYC_PER_TICK (sys_clock_hw_cycles_per_sec() / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+#define MAX_TICKS ((COUNTER_SPAN / 2 - CYC_PER_TICK) / CYC_PER_TICK)
```

Comment:
I think the max_ticks should be adjusted depending on watchdog settings, i.e. watchdog is always enabled after entering extended sleep so we need to make sure that we'll wake up before it expires

Reply:
While I tried to add this functionality I'm not sure it's needed.
If CONFIG_WDT_SMARTBOND is not set. power.c will disable watchdog after wakeup.
If watchdog is configured. Whatever task is feeding watchdog must wakeup before watchdog timer expires.
If task/work or something else is not armed to be ready before watchdog bites, even if platform is woken from sleep responsible task will not be triggered and watchdog timeout will expire anyway.
Unless I'm missing something...

Reply:
Now I see.
When watchdog is disabled it will start when PD_SYS is powered off and sleep time should not exceed this time.
Which can be RCX or RC32K based so it should be computed.

Reply:
correct, so code should be only enabled if CONFIG_PM is enabled to prevent watchdog from expiring during extended sleep period. as you said, in other case something should feed watchdog before it expires.